### ISSUE TITLE
Exercise 2.2.27.

### ISSUE THREAD

**Issue by YRFT** [reneargento/algorithms-sedgewick-wayne#177]
Here are the results that I have got:

```
N:        2 Subarray Length (percent): 50.000000%
N:        3 Subarray Length (percent): 45.555556%
N:        4 Subarray Length (percent): 33.340383%
N:        5 Subarray Length (percent): 29.833333%
N:        8 Subarray Length (percent): 20.004124%
N:        9 Subarray Length (percent): 19.351852%
N:       16 Subarray Length (percent): 11.108990%
N:       17 Subarray Length (percent): 11.127451%
N:       32 Subarray Length (percent): 5.882483%
N:       33 Subarray Length (percent): 6.414141%
N:       64 Subarray Length (percent): 3.028712%
N:       65 Subarray Length (percent): 2.961538%
N:      128 Subarray Length (percent): 1.539513%
N:      129 Subarray Length (percent): 1.543928%
N:      256 Subarray Length (percent): 0.778085%
N:      257 Subarray Length (percent): 0.761997%
N:      512 Subarray Length (percent): 0.386401%
N:      513 Subarray Length (percent): 0.378493%
N:     1024 Subarray Length (percent): 0.196712%
N:     1025 Subarray Length (percent): 0.194309%
N:     2048 Subarray Length (percent): 0.096894%
N:     2049 Subarray Length (percent): 0.097609%
N:     4096 Subarray Length (percent): 0.049283%
N:     4097 Subarray Length (percent): 0.050647%
N:     8192 Subarray Length (percent): 0.024566%
N:     8193 Subarray Length (percent): 0.022784%
N:    16384 Subarray Length (percent): 0.012666%
N:    16385 Subarray Length (percent): 0.012817%
N:    32768 Subarray Length (percent): 0.006033%
N:    32769 Subarray Length (percent): 0.005696%
N:    65536 Subarray Length (percent): 0.002794%
N:    65537 Subarray Length (percent): 0.003408%
N:   131072 Subarray Length (percent): 0.001568%
N:   131073 Subarray Length (percent): 0.001323%
N:   131074 Subarray Length (percent): 0.001526%
N:   262144 Subarray Length (percent): 0.000992%
N:   262145 Subarray Length (percent): 0.000739%
N:   262146 Subarray Length (percent): 0.000906%
N:   262147 Subarray Length (percent): 0.000858%
N:   262148 Subarray Length (percent): 0.000636%
N:   524288 Subarray Length (percent): 0.000318%
N:   524289 Subarray Length (percent): 0.000381%
N:   524290 Subarray Length (percent): 0.000191%
N:   524291 Subarray Length (percent): 0.000334%
N:   524292 Subarray Length (percent): 0.000238%
N:   524293 Subarray Length (percent): 0.000334%
N:   524294 Subarray Length (percent): 0.000620%
N:   524295 Subarray Length (percent): 0.000334%
N:   524296 Subarray Length (percent): 0.000954%
N:  1048576 Subarray Length (percent): 0.000191%
N:  1048577 Subarray Length (percent): 0.000095%
N:  1048578 Subarray Length (percent): 0.000095%
N:  1048579 Subarray Length (percent): 0.000191%
N:  1048580 Subarray Length (percent): 0.000095%
N:  1048581 Subarray Length (percent): 0.000286%
N:  1048582 Subarray Length (percent): 0.000572%
N:  1048583 Subarray Length (percent): 0.000572%
N:  1048584 Subarray Length (percent): 0.000095%
N:  1048585 Subarray Length (percent): 0.000191%
N:  1048586 Subarray Length (percent): 0.000095%
N:  1048587 Subarray Length (percent): 0.000095%
N:  1048588 Subarray Length (percent): 0.000095%
N:  1048589 Subarray Length (percent): 0.000095%
N:  1048590 Subarray Length (percent): 0.000095%
N:  1048591 Subarray Length (percent): 0.000286%
```

Which seems like f(N)=1/N.
For the probability is equal that the element is picked from left part or right part. The consumption rate of the left part and right part tend to be equal. When any part have been exhausted, the other part is almost there too, so the length of the remainder part is likely to be 1.

**Comment by reneargento** [comment]
Thanks for the contribution!
I updated the computation of the length here: https://github.com/reneargento/algorithms-sedgewick-wayne/commit/5d8738c9b1bf2d6c0b4edea95db4a2de48c0f688

In my experiments I got a different result: 36%. 
It might make sense: we want the average length of the subarray as a function of the sum of the two subarray sizes.
Following what you said about always having 1 element left on the subarray:
Most of the arrays in merges will have size 2 or 4. When the array has size 2 and 1 element is remaining on the subarray, we have 1 / 2, which is 50%. When the array has size 4, we have 1 / 4, which is 25%.
36% seems to be the average of these most common cases.
